lib/state_machine.rb

```diff
@@ -1,7 +1,162 @@
 # frozen_string_literal: true
 
 module StateMachine
-  def test
-    'test'
+  InitialStateDuplicateError = Class.new(RuntimeError)
+  StateAlreadyDefinedError = Class.new(RuntimeError)
+  InvalidStateError = Class.new(RuntimeError)
+  InvalidTransitionError = Class.new(RuntimeError)
+  GuardIsNotBooleanError = Class.new(RuntimeError)
+
+  def self.included(base)
+    base.extend(ClassMethods)
+
+    base.send(:setup)
+  end
+
+  def initialize(state_name = nil)
+    state = states.find { |s| s.name == state_name } || self.class.initial_state
+    transit(to: state)
+
+    super()
```

Comment:
Ok, you are breaking constructor for a class. if a class that includes your module expects some variables initialization variables?